src/components/general/mediaTypes/ImageViewer.tsx

```diff
@@ -0,0 +1,319 @@
+            selectAction = PointerSelectAction.EDIT;
+        } else {
+            selectAction = PointerSelectAction.HIGHLIGHT;
+        }
+
+        return selectAction;
+    }
+
+    /* Function for calculating annotation positions */
+    const CalculateAnnotationPositions = () => {
+        const visualImageAnnotations: Dict[] = [];
+
+        if (image) {
+            /* Get original size of image */
+            const imgWidth = image.width;
+            const imgHeight = image.height;
+
+            /* For each annotation, calculate the W3C pixels relative to the TDWG AC position */
+            digitalMediaAnnotations.visual.forEach((imageAnnotation: Annotation) => {
+                /* Check if the annotation is a visual one */
+                if (imageAnnotation.target.selector?.hasROI) {
+                    const ROI = imageAnnotation.target.selector.hasROI;
+
+                    const x = ROI["ac:xFrac"] * imgWidth;
+                    const y = ROI["ac:yFrac"] * imgHeight;
+                    const w = ROI["ac:widthFrac"] * imgWidth;
+                    const h = ROI["ac:heightFrac"] * imgHeight;
+
+                    visualImageAnnotations.push({
+                        id: imageAnnotation.id,
+                        "@context": 'http://www.w3.org/ns/anno.jsonld',
+                        type: 'Annotation',
+                        body: [{
+                            type: 'TextualBody',
+                            value: imageAnnotation.body.values,
+                            purpose: 'commenting',
+                            creator: {
+                                id: imageAnnotation.creator
+                            }
+                        }],
+                        target: {
+                            selector: {
+                                conformsTo: 'http://www.w3.org/TR/media-frags/',
+                                type: 'FragmentSelector',
+                                value: `xywh=pixel:${x},${y},${w},${h}`
+                            },
+                            source: digitalMedia.mediaUrl
+                        }
+                    });
+                }
+            });
+        }
+
+        return visualImageAnnotations;
+    }
+
+    /* Function for submitting an Annotation */
+    const SubmitAnnotation = (values: string[], method: string) => {
+        if (image) {
+            /* Calculate W3C pixels to TDWG AC position */
+            const coordinates = editAnnotation.target.selector.value.split(',');
+
+            const xFrac = Number(coordinates[0].replace('xywh=pixel:', '')) / image.width;
+            const yFrac = Number(coordinates[1]) / image.height;
+            const widthFrac = Number(coordinates[2]) / image.width;
+            const heightFrac = Number(coordinates[3]) / image.height;
+
+            /* Prepare new Annotation object */
+            const imageAnnotation: ImageAnnotationTemplate = {
```

Comment:
What is the difference with the above one?

Reply:
The one above is for feeding Annotorious, they use the w3 format and use a kinda different structured object.
The one below is for saving the annotation in our system, calculate the TDWG ac standard from the w3 standard (pixels to position).

Reply:
Yes thanks I forgot that annototious also uses the w3 format